### ISSUE TITLE
arm64 golang hello-wold fails to run on MacBook Air M1

### ISSUE THREAD

**Issue by ernestomedina17** [nanovms/nanos#1965]
My system info:
```
% uname -a
Darwin M1.local 22.6.0 Darwin Kernel Version 22.6.0: Wed Jul  5 22:22:52 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T8103 arm64

% go version
go version go1.21.3 darwin/arm64

% /opt/homebrew/bin/qemu-system-aarch64 -version
QEMU emulator version 8.1.2
Copyright (c) 2003-2023 Fabrice Bellard and the QEMU Project developers

% ops profile
Ops version:
Nanos version: 0.0
Qemu version: 8.1.2
OS: darwin
Arch: arm64
Virtualized: false
```

How to reproduce: 

```
% GOARCH=arm64 GOOS=linux go build .
% ops run hello -v --show-debug | head -100
 100% |█████████████████████|
Manifest:
	&{root:map[arguments:[hello] children:map[etc:map[passwd:/var/folders/9h/8ygmpcr54sn8b4g9fxvs8zr40000gn/T/hello_temp73307946/passwd resolv.conf:/var/folders/9h/8ygmpcr54sn8b4g9fxvs8zr40000gn/T/hello_temp73307946/resolv.conf ssl:map[certs:map[ca-certificates.crt:/Users/ernesto/.ops/common/ca-certificates.crt]]] hello:hello lib:map[x86_64-linux-gnu:map[libnss_dns.so.2:/Users/ernesto/.ops/common/libnss_dns.so.2]] proc:map[sys:map[kernel:map[hostname:/var/folders/9h/8ygmpcr54sn8b4g9fxvs8zr40000gn/T/hello_temp73307946/hostname]]]] environment:map[IMAGE_NAME:hello NANOS_VERSION:0.0 OPS_VERSION: PWD:/ USER:root] program:/hello] boot:map[children:map[kernel:/Users/ernesto/.ops/0.1.48-arm/kernel.img]] targetRoot: klibHostDir:/Users/ernesto/.ops/0.1.48/klibs}
running local instance
booting /Users/ernesto/.ops/images/hello ...
qemu-system-aarch64 -machine virt -machine gic-version=2 -machine highmem=off -kernel /Users/ernesto/.ops/0.1.48-arm/kernel.img -device virtio-blk-pci,drive=hd0 -semihosting -m 1G -device virtio-rng-pci -accel hvf -cpu host,-rdtscp -no-reboot -cpu host -drive file=/Users/ernesto/.ops/images/hello,format=raw,if=none,id=hd0 -device virtio-net,netdev=n0,mac=9e:bb:17:35:8e:7a -netdev user,id=n0 -display none -serial stdio

no fault handler for frame
     frame: 00ff800040e30000
      type: kernel
      spsr: 00000000200003c5
       esr: 0000000096000047 data abort in el1 write
       far: 00ff000100025000
       elr: ffffffff80077d18
       far: 00ff000100025000
active_cpu: 0000000000000000
 stack top: 00ff800040e37ff0

        x0: 00ff000100025000
        x1: 0000000000000000
        x2: 0000000000000210
        x3: 0000000000000000
        x4: 00ff000100025000
        x5: 00ff000100025000
        x6: 0000000000000041
        x7: 00ff000100025210
        x8: 0000000000000042
        x9: 0000000000000042
       x10: 0000000000000000
       x11: 0000000000000026
       x12: 0000000000000001
       x13: 00ff800041003500
       x14: 00ff800041003500
       x15: 0000000000000000
       x16: ffffffff80072f30
       x17: 00ff800040e30100
       x18: 00ff800040e25000
       x19: 00ff000100025000
       x20: 00ff800040ee0000
       x21: 00ff800040e2a000
       x22: ffffffff80141000
       x23: 0000000000000001
       x24: 00ff800041a05000
       x25: 00ff800040e25000
       x26: ffffffff80141000
       x27: 00ff800040e60008
       x28: 00000000000003c0
       x29: 00ff800040e37d60
       x30: ffffffff80028524
        sp: 00ff800040e37d60
      fpcr: 0000000000008000

frame trace:
loaded klibs:
stack trace:

frame 0x00ff800040e30000 already full
frame trace:
loaded klibs:

assertion !frame_is_full(ctx->frame) failed at /home/eyberg/go/src/github.com/nanovms/nanos/src/kernel/mutex.c:109 (IP 0xffffffff80048ff4)  in mutex_lock_internal(); halt`

**Comment by eyberg** [comment]
btw - have you tried w/nightly (eg: 'ops run -n hello') or another nanos version (via --nanos-version) ?

**Comment by francescolavra** [comment]
Thanks for reporting this issue @ernestomedina17. It's fixed by #1966
